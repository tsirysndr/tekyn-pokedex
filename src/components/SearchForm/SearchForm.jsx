import React from "react";
import PropTypes from "prop-types";
import { View } from "react-native";

export default function Infos(props) {
  return <View></View>;
}
