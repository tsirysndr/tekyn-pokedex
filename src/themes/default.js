export const theme = {
  // Colors
  colors: {
   normal: "#A8A87B",
   fire: "#EE803B",
   water: "#6A92ED",
   grass: "#7BC757",
   electrique: "#F7CF43",
   ice: "#9AD8D8",
   fight: "#BE322E",
   poison: "#9B4A99",
   ground: "#DFBF6E",
   flight: "#A893ED",
   psychic: "#EC5B89",
   bug: "#A8B732",
   rock: "#B89F41",
   ghost: "#705A97",
   dark: "#705849",
   dragon: "#7043F4",
   steel: "#B8B9CF",
   fairy: "#EFB7BD",
  }
}