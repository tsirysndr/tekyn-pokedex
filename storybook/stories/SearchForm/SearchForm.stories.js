import React from "react";
import SearchForm from "@Components/SearchForm";
import { storiesOf } from "@storybook/react-native";
import CenterView from "../CenterView";

storiesOf("SearchForm", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("default", () => <SearchForm />);
