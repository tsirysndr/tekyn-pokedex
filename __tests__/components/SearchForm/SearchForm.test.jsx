import SearchForm from "@Components/SearchForm";
import { ThemeProvider } from "@emotion/react";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import { theme } from "@Themes/default";

test("renders the SearchForm", () => {
  const onSearch = jest.fn();
  const onShufflePokemon = jest.fn();
  const tree = render(
    <ThemeProvider theme={theme}>
      <SearchForm onSearch={onSearch} onShufflePokemon={onShufflePokemon} />
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});

test("enter pokemon name and click SearchButton", async () => {
  const onSearch = jest.fn();
  const onShufflePokemon = jest.fn();
  const { getByPlaceholderText, getAllByRole } = render(
    <ThemeProvider theme={theme}>
      <SearchForm onSearch={onSearch} onShufflePokemon={onShufflePokemon} />
    </ThemeProvider>
  );
  fireEvent.changeText(getByPlaceholderText("name or id"), "Pikachu");
  const buttons = getAllByRole("button");
  fireEvent.press(buttons[0]);
  await waitFor(() => expect(onSearch).toBeCalledWith({ nameOrId: "Pikachu" }));
});

test("click SearchButton, skip if input is empty", () => {
  const onSearch = jest.fn();
  const onShufflePokemon = jest.fn();
  const { getAllByRole } = render(
    <ThemeProvider theme={theme}>
      <SearchForm onSearch={onSearch} onShufflePokemon={onShufflePokemon} />
    </ThemeProvider>
  );
  const buttons = getAllByRole("button");
  fireEvent.press(buttons[0]);
  expect(onSearch).not.toBeCalled();
});

test("click ShuffleButton", () => {
  const onSearch = jest.fn();
  const onShufflePokemon = jest.fn();
  const { getAllByRole } = render(
    <ThemeProvider theme={theme}>
      <SearchForm onSearch={onSearch} onShufflePokemon={onShufflePokemon} />
    </ThemeProvider>
  );
  const buttons = getAllByRole("button");
  fireEvent.press(buttons[1]);
  expect(onShufflePokemon).toBeCalled();
});
