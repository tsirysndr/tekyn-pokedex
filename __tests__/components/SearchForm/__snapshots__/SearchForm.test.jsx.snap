// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SearchForm 1`] = `
<View
  height={317}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.8)",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      },
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  width={300}
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 50,
          "marginBottom": 30,
          "width": 50,
        },
        undefined,
      ]
    }
  />
  <View>
    <TextInput
      fontsLoaded={true}
      onChangeText={[Function]}
      placeholder="name or id"
      placeholderTextColor="#fff"
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "color": "#000",
            "fontFamily": "Roboto_700Bold",
            "height": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 250,
          },
          undefined,
        ]
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 25,
          "width": 250,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F4A261",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "width": 150,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F4A261",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "width": 45,
        }
      }
    >
      <RNSVGSvgView
        align="none"
        bbHeight={18}
        bbWidth={18}
        fill="none"
        focusable={false}
        height={18}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 18,
              "width": 18,
            },
          ]
        }
        vbHeight={25}
        vbWidth={30}
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M29.435 19.702 23.254 25v-3.532h-1.04c-3.285 0-5.704-1.472-7.701-3.42a62.845 62.845 0 0 0 2.118-2.673l.445-.586c1.477 1.528 3.066 2.558 5.138 2.558h1.04v-2.944l6.18 5.3ZM0 7.928h3.08c1.954 0 3.478.918 4.884 2.301.231-.297.465-.606.701-.915a70.015 70.015 0 0 1 1.842-2.35C8.562 5.149 6.217 3.807 3.08 3.807H0v4.122ZM23.254 0v3.512h-1.04c-5.437 0-8.502 4.03-11.207 7.586-2.432 3.193-4.533 5.954-7.927 5.954H0v4.121h3.08c5.437 0 8.502-4.027 11.207-7.582 2.432-3.197 4.533-5.958 7.926-5.958h1.04v2.965l6.182-5.3L23.254 0Z"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
